src/components/dropdown-menu/dropdown-menu.jsx

```diff
@@ -0,0 +1,205 @@
+		onOpenChange: setIsOpen,
+		placement,
+		strategy: 'absolute',
+		middleware: [
+			offset( offsetValue ),
+			flip( { boundary } ),
+			shift( { boundary } ),
+		],
+		whileElementsMounted: autoUpdate,
+	} );
+
+	const click = useClick( context );
+	const dismiss = useDismiss( context );
+	const role = useRole( context, { role: 'menu' } );
+
+	const { getReferenceProps, getFloatingProps } = useInteractions( [
+		click,
+		dismiss,
+		role,
+	] );
+
+	const { isMounted, styles } = useTransitionStyles( context, {
+		duration: 150,
+		initial: { opacity: 0, scale: 0.95 },
+		open: { opacity: 1, scale: 1 },
+		close: { opacity: 0, scale: 0.95 },
+	} );
+
+	const toggleMenu = () => setIsOpen( ( prev ) => ! prev );
+
+	const handleClose = () => setIsOpen( false );
+
+	return (
+		<DropdownMenuContext.Provider value={ { handleClose } }>
+			<div className={ cn( 'relative inline-block', className ) }>
+				<div
+					ref={ refs.setReference }
+					onClick={ toggleMenu }
+					role="button"
+					tabIndex={ 0 }
+					{ ...getReferenceProps() }
+					className="cursor-pointer"
+				>
+					{ React.Children.map( children, ( child ) => {
+						if (
+							child.type?.displayName === 'DropdownMenu.Trigger'
+						) {
+							return child;
+						}
+						return null;
+					} ) }
+				</div>
+
+				{ isMounted && (
+					<FloatingPortal
+						id={ dropdownPortalId }
+						root={ dropdownPortalRoot }
+					>
+						<div
+							ref={ refs.setFloating }
+							style={ {
+								...floatingStyles,
+								...styles,
+							} }
+							{ ...getFloatingProps() }
+						>
+							{ React.Children.map( children, ( child ) => {
+								if (
+									child.type?.displayName ===
+									'DropdownMenu.Content'
+								) {
+									return child;
+								}
+								return null;
+							} ) }
+						</div>
+					</FloatingPortal>
+				) }
+			</div>
+		</DropdownMenuContext.Provider>
+	);
+};
+
+DropdownMenu.displayName = 'DropdownMenu';
+
+const DropdownMenuTrigger = React.forwardRef( ( { children, className }, ref ) => (
+	<div ref={ ref } role="button" tabIndex={ 0 } className={ className }>
+		{ children }
+	</div>
+) );
+
+DropdownMenuTrigger.displayName = 'DropdownMenu.Trigger';
+
+const DropdownMenuContent = ( { children, className, ...props } ) => {
+	return (
+		<div
+			className={ cn(
+				'border border-solid border-border-subtle rounded-md shadow-lg overflow-hidden',
+				className
+			) }
+		>
+			<Menu { ...props }>{ children }</Menu>
+		</div>
+	);
+};
+DropdownMenuContent.displayName = 'DropdownMenu.Content';
+
+const DropdownMenuList = ( { children, ...props } ) => {
+	const { handleClose } = useDropdownMenuContext();
+
+	return (
+		<Menu.List { ...props }>
+			{ React.Children.map( children, ( child ) => {
+				if ( React.isValidElement( child ) ) {
+					return React.cloneElement( child, {
+						onClick: callAll( child.props?.onClick, handleClose ),
+					} );
+				}
+				return child;
+			} ) }
+		</Menu.List>
+	);
+};
```

Comment:
We don't need to attach close dropdown handler to the children as we are doing it from the `DropdownMenuItem` component.

```suggestion
const DropdownMenuList = ( { children, ...props } ) => {
	return (
		<Menu.List { ...props }>
			{ children }
		</Menu.List>
	);
};
```